[ENH] Add `Commenting on a PR` to CONTRIBUTING.md (#490)
bids-standard/bids-specification: CONTRIBUTING.md

    In addition to the specification itself, we are dedicated to creating a healthy community.
    These issues highlight pieces of work or discussions around how we can support our members and make it easier to contribute.

## Commenting on a pull request

Our primary method of adding to or enhancing BIDS occurs in the form of [pull requests](https://help.github.com/articles/about-pull-requests/).

> Extentions to BIDS (or [BEPs](https://docs.google.com/document/d/1pWmEEY-1-WuwBPNy5tDAxVJYQ9Een4hZJM06tQZg8X4/edit)) are submitted as pull requests.

Reviewer: ```suggestion
BIDS Extension Proposals ([BEPs](https://docs.google.com/document/d/1pWmEEY-1-WuwBPNy5tDAxVJYQ9Een4hZJM06tQZg8X4/)) are submitted as pull requests.
```

Commenting on pull requests is an important way of participating in the BIDS community.

Please follow the instructions for how to comment on a pull request.

### Navigating to open pull requests

> Starting from our [BIDS-Specification GitHub](https://github.com/bids-standard/bids-specification), please click on `pull requests`

Reviewer: ```suggestion
Starting from our [BIDS-Specification GitHub](https://github.com/bids-standard/bids-specification), please click on `Pull requests`
```

Descriptions are important for reviewers and the community to gain context into what the pull request is achiving.

![BIDS-pr](commenting_images/BIDS_pr.png "BIDS_pr")

### Generally commenting on a pull request

Please scroll to the bottom of the pull request to find a comment box.
These comments will address the pull request in general.
Next, we will go over how to comment on specific lines of the pull request.

![BIDS-comment](commenting_images/BIDS_comment.png "BIDS-comment")

### Specific comments on a pull request

Please click on the `Files changed` tab from the pull request page.
This tab displays what this pull request seeks to change.
The green text is being added while the red is being removed.
To comment on a specific line, please click on the `plus sign` (pictured below).
To comment on several consecutive lines, please click and drag until the yellow highlight captures the lines of interest.

![BIDS-specific-comment](commenting_images/BIDS_file_comment.png "BIDS-specific-comment")

#### Suggesting text

When you are making specific line comments, you may also suggest a text change.
To do this, please first select the line(s) you would like to make a suggestion.

> Then please click the plus/negative icon in the comment box (pictured below).

Reviewer: ```suggestion
Then please click the plus/minus icon in the comment box (pictured below).
```

![BIDS-suggest-box](commenting_images/BIDS_suggest.png "BIDS-suggest")

> Once the button is clicked the highlighted text will be copied into the comment box and formatted as [Markdown](https://www.markdownguide.org/basic-syntax/).

Reviewer: ```suggestion
Once the button is clicked the highlighted text will be copied into the comment box and formatted as
a [Markdown code block](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks).
```

![BIDS-suggest-text](commenting_images/BIDS_suggest_text.png "BIDS-suggest-box")

> As you work on the suggestion you may preview the comment by clicking the `preview` button. 

Reviewer: ```suggestion
As you work on the suggestion you may preview the comment by clicking the `Preview` button. 
```

### Navigating to open pull requests

Starting from our [BIDS-Specification GitHub](https://github.com/bids-standard/bids-specification), please click on `Pull requests`

![BIDS-mainpage](commenting_images/BIDS_GitHub_mainpage.png "BIDS_GitHub_mainpage")

### Selecting an open pull request

In this example we will be navigating to our [BIDS common derivatives pull request](https://github.com/bids-standard/bids-specification/pull/265).

![BIDS-pr-list](commenting_images/BIDS_pr_list.png "BIDS_pr_list")

### Pull request description

Upon opening the pull request we see a detailed description of what this pull request is seeking to address.
Descriptions are important for reviewers and the community to gain context into what the pull request is achiving.

![BIDS-pr](commenting_images/BIDS_pr.png "BIDS_pr")

### Generally commenting on a pull request

Please scroll to the bottom of the pull request to find a comment box.
These comments will address the pull request in general.
Next, we will go over how to comment on specific lines of the pull request.

![BIDS-comment](commenting_images/BIDS_comment.png "BIDS-comment")

### Specific comments on a pull request

> Please click on the `Files changed` tab from the pull request page.
> This tab displays what this pull request seeks to change.
> The green text is being added while the red is being removed.
> To comment on a specific line, please click on the `plus sign` (pictured below).
> To comment on several consecutive lines, please click and drag until the yellow highlight captures the lines of interest.

Reviewer 2: ```suggestion
The proposed changes to the text of the specification can be seen in the "Files changed" tab.
Proposed additions are displayed on a green background with a `+` before each added line.
Proposed deletions are displayed on a red background with a `-` before each removed line.
To comment on a specific line, hover over it, and click the blue plus sign (pictured below).
Multiple lines can be selected by clicking and dragging the plus sign.
```

![BIDS-specific-comment](commenting_images/BIDS_file_comment.png "BIDS-specific-comment")

#### Suggesting text

> When you are making specific line comments, you may also suggest a text change.
> To do this, please first select the line(s) you would like to make a suggestion.
> Then please click the plus/minus icon in the comment box (pictured below).

Reviewer 2: ```suggestion
Comments on lines can contain "suggestions", which allow you to propose specific wording for consideration.
To make a suggestion, click the plus/minus (±) icon in the comment box (pictured below).
```

![BIDS-suggest-box](commenting_images/BIDS_suggest.png "BIDS-suggest")

Once the button is clicked the highlighted text will be copied into the comment box and formatted as
a [Markdown code block](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks).

![BIDS-suggest-text](commenting_images/BIDS_suggest_text.png "BIDS-suggest-box")

> As you work on the suggestion you may preview the comment by clicking the `Preview` button. 
> 
> ![BIDS-suggest-preview](commenting_images/BIDS_suggest_preview.png "BIDS-suggest-preview")
> 
> When the text has changed, the preview will identify the differences.
> 
> ![BIDS-suggest-change](commenting_images/BIDS_suggest_change.png "BIDS-suggest-change")

Reviewer 2: This is cutting it a bit fine, IMO. I think one image showing the preview and the highlighting gets the idea across plenty.

```suggestion
The "Preview" tab in the comment box will show your suggestion as it will be rendered.
The "Suggested change" box will highlight the differences between the original text and your suggestion.

![BIDS-suggest-change](commenting_images/BIDS_suggest_change.png "BIDS-suggest-change")
```

> When you have completed your suggestion, you may either `Start a review` or `Add single comment`.
> `Start a review` allows you to write more comments and submit them as a batch of feedback.
> As opposed to `Add single comment` which immediately submits the comment.

Reviewer 2: ```suggestion
A comment may be submitted on its own by clicking "Add single comment".
Several comments may be grouped by clicking "Start a review".
As more comments are written, accept them with "Add review comment", and submit your review comments
as a batch by clicking the "Finish your review" button.
```
